src/components/banner/CdrBanner.vue

```diff
@@ -1,36 +1,69 @@
+/* eslint-disable no-unused-vars */
 <template>
   <div :class="[style[baseClass], style[typeClass]]">
-    <slot />
+    <div :class="[style['cdr-banner__wrapper'], style[prominenceClass]]">
+      <div :class="[style['cdr-banner__main']]">
+        <div
+          v-if="hasIconLeft"
+          :class="[style['cdr-banner__icon-left']]"
+        >
+          <slot name="icon-left" />
+        </div>
+        <span :class="[style['cdr-banner__message']]">
+          <slot />
+        </span>
+        <div
+          v-if="hasIconRight"
+          :class="[style['cdr-banner__icon-right']]"
+        >
+          <slot name="icon-right" />
+        </div>
+      </div>
+      <div
+        v-if="hasMessageBody"
+        :class="[style['cdr-banner__message-body']]"
+      >
+        <slot name="message-body" />
+      </div>
+    </div>
+    <div
+      v-if="hasInfoAction"
+      :class="[style['cdr-banner__info-action']]"
+    >
+      <slot name="info-action" />
+    </div>
   </div>
 </template>
 
-<script>
-import { defineComponent, useCssModule, computed } from 'vue';
+<script setup>
+import {
+  defineProps, useCssModule, computed, useSlots,
+} from 'vue';
 import propValidator from '../../utils/propValidator';
 import { buildClass } from '../../utils/buildClass';
 
-export default defineComponent({
-  name: 'CdrBanner',
-  props: {
-    type: {
-      type: String,
-      validator: (value) => propValidator(
-        value,
-        ['info', 'warning', 'success', 'error'],
-      ),
-      default: 'info',
-    },
-  },
-  setup(props) {
-    const baseClass = 'cdr-banner';
-    const typeClass = computed(() => props.type && buildClass(baseClass, props.type));
-    return {
-      style: useCssModule(),
-      baseClass,
-      typeClass,
-    };
+const ctx = useSlots();
```

Comment:
If I'm reading this right, the `useSlots()` method here returns an array of slotted elements? So when we define the constants `hasIconLeft`, `hasIconRight`, etc these refer to the elements themselves?

Reply:
It's a helper runtime function that returns the slots object (i.e. what slots the component is using at runtime). It's essentially the equivalent of this using JSX syntax https://github.com/rei/rei-cedar/blob/next/src/components/banner/CdrBanner.jsx#L50

Reply:
Got it. That makes sense